astropy/time/formats.py

```diff
@@ -1259,7 +1346,64 @@ def set_jds(self, val1, val2):
 
         jd1, jd2 = erfa.dtf2d(self.scale.upper().encode('ascii'),
                               *iterator.operands[1:])
-        self.jd1, self.jd2 = day_frac(jd1, jd2)
+        jd1, jd2 = day_frac(jd1, jd2)
+
+        return jd1, jd2
+
+    def get_jds_fast(self, val1, val2):
+        """Use fast C parser to parse time strings in val1 and get jd1, jd2"""
+        # Handle bytes or str input and flatten down to a single array of uint8.
+        char_size = 4 if val1.dtype.kind == 'U' else 1
+        val1_str_len = int(val1.dtype.itemsize // char_size)
+        chars = val1.ravel().view(np.uint8)
+
+        if char_size == 4:
+            # Check that this is pure ASCII
+            status = self.lib_parse_time.check_unicode(chars, len(chars) // 4)
+            if status != 0:
+                raise ValueError('input is not pure ASCII')
+            # It might be possible to avoid this copy with cleverness in
+            # parse_times.c but leave that for another day.
+            chars = chars[::4]
+        chars = np.ascontiguousarray(chars)
+
+        # Pre-allocate output components
+        n_times = len(chars) // val1_str_len
+        year = np.zeros(n_times, dtype=np.intc)
+        month = np.zeros(n_times, dtype=np.intc)
+        day = np.zeros(n_times, dtype=np.intc)
+        hour = np.zeros(n_times, dtype=np.intc)
+        minute = np.zeros(n_times, dtype=np.intc)
+        second = np.zeros(n_times, dtype=np.double)
+
+        # Set up parser parameters as numpy arrays for passing to C parser
+        pars = self.fast_parser_pars
+        delims = np.array(pars['delims'], dtype=np.uint8)
```

Comment:
Would it be an idea to create a structured array? Those may to C-side structs, so are easy to use and it would mean only passing one item on.

If one goes that route, one could also create a structured output array.

Reply:
I have the structured output array working and this is much nicer. For inputs I'm leaving them separate (see later comment).